runners: jlink: Add support for big endian device

sigmundklaa opened the pull request:
Adds the --endian option, setting endianness for both the JLink and the GDB server.

Documented at:
https://wiki.segger.com/J-Link_Commander#LE
https://wiki.segger.com/J-Link_GDB_Server#-endian

henrikbrixandersen commented on scripts/west_commands/runners/jlink.py:
```diff
@@ -243,6 +247,7 @@ def do_run(self, command, **kwargs):
                        '-speed', self.speed,
                        '-device', self.device,
                        '-silent',
+                       '-endian', self.endian,
```
Can we set this automatically based on `CONFIG_BIG_ENDIAN`/`CONFIG_LITTLE_ENDIAN`?

sigmundklaa replied:
Ah yes, that is a good idea. What is the best way of doing that? Using `zcmake.CMakeCache`, or is there another way?

henrikbrixandersen replied:
You should be able to use `self.build_conf.getboolean('CONFIG_BIG_ENDIAN')` within the runner. Check the other runners for examples on how this can be done.

henrikbrixandersen commented:
@sigmundklaa You need to fix your Git commit message:

> 1: UC2 Signed-off-by: must have a full name
> Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details
> 
> aba8cc45349f6249a74b1e926b0e8799582acbc9: author email (sigmundklaa <sigmundklaa@outlook.com>) does not follow the syntax: First Last <email>.

sigmundklaa committed:
runners: jlink: Add support for big endian device

Sets the endianness of the JLink and the GDB Server based on if
`CONFIG_BIG_ENDIAN` is set.

Doc on JLink LE/BE option:
https://wiki.segger.com/J-Link_Commander#LE

Doc on GDB server -endian option:
https://wiki.segger.com/J-Link_GDB_Server#-endian

Signed-off-by: Sigmund Klåpbakken <sigmundklaa@outlook.com>
